volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,165 @@
+        ]
+
+    def _generator(self):
+        layer = self.context.layers[self.config['primary']]
+
+        # Yara Rule to scan for MFT Header Signatures
+        rules = yarascan.YaraScan.process_yara_options({'yara_rules': '/FILE0|FILE\*|BAAD/'})
+
+        # Read in the Symbol File
+        symbol_table = intermed.IntermediateSymbolTable.create(context = self.context,
+                                                               config_path = self.config_path,
+                                                               sub_path = "windows",
+                                                               filename = "mft",
+                                                               class_types = {
+                                                                   'FILE_NAME_ENTRY': mft.MFTFileName,
+                                                                   'MFT_ENTRY': mft.MFTEntry
+                                                               })
+
+        # get each of the individual Field Sets
+        mft_object = symbol_table + constants.BANG + "MFT_ENTRY"
+        attribute_object = symbol_table + constants.BANG + "ATTRIBUTE"
+        header_object = symbol_table + constants.BANG + "ATTR_HEADER"
+        si_object = symbol_table + constants.BANG + "STANDARD_INFORMATION_ENTRY"
+        fn_object = symbol_table + constants.BANG + "FILE_NAME_ENTRY"
+
+        # Get the Enums
+        attr_types = self.context.symbol_space.get_enumeration(symbol_table + constants.BANG + "AttrTypeEnum")
+        namespace_enum = self.context.symbol_space.get_enumeration(symbol_table + constants.BANG + "NameSpaceEnum")
+        mft_flags = self.context.symbol_space.get_enumeration(symbol_table + constants.BANG + "MFTFlagsEnum")
+        permission_flags = self.context.symbol_space.get_enumeration(symbol_table + constants.BANG +
+                                                                     "PermissionFlagEnum")
+
+        # Scan the layer for Raw MFT records and parse the fields
+        for offset, _rule_name, _name, _value in layer.scan(context = self.context,
+                                                            scanner = yarascan.YaraScanner(rules = rules)):
+            try:
+                mft_record = self.context.object(mft_object, offset = offset, layer_name = layer.name)
+                # We will update this on each pass in the next loop and use it as the new offset.
+                attr_base_offset = mft_record.FirstAttrOffset
+
+                attr_header = self.context.object(header_object,
+                                                  offset = offset + attr_base_offset,
+                                                  layer_name = layer.name)
+
+                # There is no field that has a count of Attributes
+                # Keep Attempting to read attributes until we get an invalid attr_header.AttrType
+                while attr_header.AttrType in attr_types.choices.values():
```

Comment:
Sorry, there is actually an `attr_header.is_valid_choice()` (https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/objects/__init__.py#L492) function, and it will make it easier for us to change if there becomes a more efficient way of doing it...